src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs

```diff
@@ -1,138 +1,146 @@
 using Edge = Autodesk.DesignScript.Geometry.Edge;
 using Face = Autodesk.DesignScript.Geometry.Face;
-using ProtoPt = Autodesk.DesignScript.Geometry.Point;
 using Solid = Autodesk.DesignScript.Geometry.Solid;
 using Surface = Autodesk.DesignScript.Geometry.Surface;
 
 namespace Revit.GeometryConversion
 {
-   [SupressImportIntoVM]
-   public static class DynamoToRevitBRep
-   {
-       /// <summary>
-       /// this method attempts to construct a BRep from a closed solid.
-       /// </summary>
-       /// <param name="sol"></param>
-       /// <param name="performHostUnitConversion"></param>
-       /// <returns></returns>
-       public static GeometryObject ToRevitType(Solid sol,
-          bool performHostUnitConversion = true)
-       {
-            var geometry = ToRevitType(sol, performHostUnitConversion, BRepType.Solid);
-
-           if (geometry == null)
-           {
-               // An unexpected failure occurred when attempting to convert the solid into a Revit BRep.
-               throw new Exception(Properties.Resources.DynamoSolidToRevitBRepFailure);
-           }
+    [SupressImportIntoVM]
+    public static class DynamoToRevitBRep
+    {
+        /// <summary>
+        /// this method attempts to construct a BRep from a closed solid.
+        /// </summary>
+        /// <param name="sol"></param>
+        /// <param name="performHostUnitConversion"></param>
+        /// <param name="materialId"></param>
+        /// <returns></returns>
+        public static GeometryObject ToRevitType(Solid sol,
+          bool performHostUnitConversion = true,
+          ElementId materialId = null)
+        {
+            var geometry = ToRevitType(sol, performHostUnitConversion, BRepType.Solid, materialId);
+
+            if (geometry == null)
+            {
+                // An unexpected failure occurred when attempting to convert the solid into a Revit BRep.
+                throw new Exception(Properties.Resources.DynamoSolidToRevitBRepFailure);
+            }
 
             return geometry;
-       }
-
-       /// <summary>
-       /// this method attempts to construct a BRep from a surface.
-       /// </summary>
-       /// <param name="surf"></param>
-       /// <param name="performHostUnitConversion"></param>
-       /// <returns></returns>
-       public static GeometryObject ToRevitType(Surface surf,
-          bool performHostUnitConversion = true)
-       {
-            var geometry = ToRevitType(surf, performHostUnitConversion, BRepType.OpenShell);
+        }
+
+        /// <summary>
+        /// this method attempts to construct a BRep from a surface.
+        /// </summary>
+        /// <param name="surf"></param>
+        /// <param name="performHostUnitConversion"></param>
+        /// <param name="materialId"></param>
+        /// <returns></returns>
+        public static GeometryObject ToRevitType(Surface surf,
+          bool performHostUnitConversion = true,
+          ElementId materialId = null)
```

Comment:
Add material id param as an optional param the end in order to maintain backward compability.